universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramSummaryView.java

```diff
@@ -0,0 +1,151 @@
+package views.applicant;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.br;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.h1;
+import static j2html.TagCreator.header;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import controllers.applicant.routes;
+import j2html.tags.ContainerTag;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Arrays;
+import play.mvc.Http.HttpVerbs;
+import play.mvc.Http.Request;
+import play.twirl.api.Content;
+import services.applicant.SummaryData;
+import views.BaseHtmlView;
+import views.components.LinkElement;
+import views.style.Styles;
+
+public final class ApplicantProgramSummaryView extends BaseHtmlView {
+
+  private final ApplicantLayout layout;
+
+  @Inject
+  public ApplicantProgramSummaryView(ApplicantLayout layout) {
+    this.layout = checkNotNull(layout);
+  }
+
+  /**
+   * Renders a summary of all of the applicant's data for a specific application. Data includes:
+   * Program Id, Applicant Id - Needed for link context (submit & edit) Question Data for each
+   * question: - question text - answer text - block id (for edit link)
+   */
+  public Content render(
+      Request request,
+      Long applicantId,
+      Long programId,
+      String programTitle,
+      ImmutableList<SummaryData> data) {
+    ContainerTag headerTag = renderHeader(100);
```

Comment:
(blocking) can you add a TODO here so don't lose track of this `100` placeholder?

Reply:
The 100 will always be true here. It's my understanding that % complete is based on position in the application, and since summary is always the last block whenever we arrive here we're at 100% done. It may be different if we want to show % of questions answered, but I dunno...

Reply:
Ahh gotcha. This is the first time I've seen anything percentage related 😁 . Can it just be a constant than? That'll make it a bit more readable.

Reply:
When I make a broader pass on the applicant styling I'll the percent complete bar portion to a shared location and do a more thorough cleanup then.